### PR TITLE
[FIX] product_variant_configurator: returning domains in onchange doesn't work anymore

### PR DESCRIPTION
**Pull request by jdoutreloux**

### CODE REVIEW

File: product_variant_configurator/models/product_configurator.py [hunk 1]
```diff
@@ -52,6 +57,19 @@ def _compute_can_be_created(self):
             )
             rec.price_extra = sum(rec.mapped("product_attribute_ids.price_extra"))
 
+    @api.depends("product_tmpl_id", "product_attribute_ids")
+    def _compute_product_id_configurator_domain(self):
+        product_obj = self.env["product.product"]
+        for rec in self:
+            if not rec.product_tmpl_id._origin:
+                # no product template: allow any product
+                rec.product_id_configurator_domain = []
```
Comment (ThomasBinsfeld) [inline product_variant_configurator/models/product_configurator.py]:
```suggestion
                rec.product_id_configurator_domain = TRUE_DOMAIN
```

from odoo.osv.expression

File: product_variant_configurator/models/product_configurator.py [hunk 2]
```diff
@@ -52,6 +57,19 @@ def _compute_can_be_created(self):
...
+            else:
+                domain, cont = product_obj._build_attributes_domain(
```
Comment (ThomasBinsfeld) [inline product_variant_configurator/models/product_configurator.py]:
```suggestion
                domain, _cont = product_obj._build_attributes_domain(
```

Prefix unused variable with `_`

File: product_variant_configurator/tests/test_product_variant_configurator.py [hunk 3]
```diff
@@ -285,21 +285,17 @@ def test_get_product_description(self):
...
         )
 
-    def test_onchange_product_tmpl_id(self):
+    def test_compute_product_id_configurator_domain(self):
         product = self.product_product.new(
             {"name": "Test product", "product_tmpl_id": self.product_template_yes.id}
         )
         product.product_tmpl_id = self.product_template_empty_yes
-        res = product._onchange_product_tmpl_id_configurator()
+        # res = product._onchange_product_tmpl_id_configurator()
```
Comment (ThomasBinsfeld) [inline product_variant_configurator/tests/test_product_variant_configurator.py]: Remove or restore commented code.

**Review by ThomasBinsfeld — CHANGES_REQUESTED** [review]
Please fix pre-commit

**Review by ThomasBinsfeld — APPROVED** [review]

**Review by FrancoMaxime — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]

**Pull request merged**